tests/Gitonomy/Git/Tests/RepositoryTest.php

```diff
@@ -54,6 +54,22 @@ public function testGetDescription($repository)
     {
         $this->assertSame("Unnamed repository; edit this file 'description' to name the repository.\n", $repository->getDescription());
     }
+    
+    /*
+     * @dataProvider provideFoobar
+     */
+    public function testPull($repository)
+    {
+        $this->assert($repository->pull());
```

Comment:
Can you test the output from the git process?

Reply:
technically yes, but I would have to write some kind parser as the output would differ per execution. This is because git list what files are pulled. That is why I return the repo from the method instead of a bool for succes. I guess this is what makes the patch rudimentary 😛

Reply:
No need to be so rigorous. Just test the string contains some substring. Probably good enough.

Reply:
Maybe just leave this test as `assertIsBool($repository->pull())`?